src/SwiftPackage.ts

```diff
@@ -150,9 +151,12 @@ export class SwiftPackage implements PackageContents {
      */
     static async loadPackage(folder: vscode.Uri): Promise<PackageContents | null | undefined> {
         try {
-            let { stdout } = await execSwift(["package", "describe", "--type", "json"], {
-                cwd: folder.fsPath,
-            });
+            let { stdout } = await execSwift(
+                ["package", "describe", "--type", "json", ...swiftpmSDKFlags()],
```

Comment:
If the sdk flags need added to every call of `swift` why not add them in the `execSwift` instead of at every call site.

Reply:
SwiftPM and Swift Driver use different styles of SDK flag, so we need to distinguish them from each other. Also, there’re some simple commands (like `swift package clean` and `swift --version`) that doesn’t depend on SDK.

In the long run, we should distinguish host SDK and target SDK, which should be passed accordingly to tasks.

Reply:
I just feel there must be a cleaner way to do this. Anyone adding new calls to `swift` will need to remember to add these.

Reply:
I tried to do so, but unfortunately should give up because of some bugs in SwiftPM (related to Argument Parser, I guess) that requires certain order of arguments, so we cannot simply append it to the end.

Reply:
I will keep the WIP branch until the issue is addressed.

Reply:
I managed to land it in the latest commit👀 Hopefully this won’t trigger the `swift-package` bug.